.github/workflows/daily-e2e-tests-libvirt.yaml

```diff
@@ -0,0 +1,144 @@
+# (C) Copyright Confidential Containers Contributors 2023.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Daily run the e2e tests for libvirt.
+---
+name: daily e2e tests for libvirt
+
+on:
+  schedule:
+  # Runs "at 04:15(UTC time) every day" (see https://crontab.guru)
+  # will base on default branch `main`
+    - cron: '15 4 * * *'
+  workflow_dispatch:
+
+env:
+    # cloud-api-adaptor image registry
+    E2E_IMG_REGISTRY: ghcr.io/${{ github.repository_owner }}
+    # cloud-api-adaptor: image release tag
+    E2E_IMG_RELEASE_TAG: latest
+    # cloud-api-adaptor image dev tag
+    E2E_IMG_DEV_TAG: latest-dev
```

Comment:
I can see that this is actually a copy of the `e2e_on_pull` with a different tags, without cloning the repo and without authorization. Wouldn't it be better to turn that to a `workflow_call` making some steps optional based on a variable?

Reply:
You are right. At first I tried to refactor to `e2e_on_pull` in order to make it callable (generic)  but I faced some limitations on the github actions syntax. Let me have a 2nd look...